### PR TITLE
Add an option for a different export path

### PR DESCRIPTION
**Pull request by Marr11317**
1) refactor the naming filenames to be more easily understandable
2) add the option to export in a different folder
3) add verification that the score was able to be exported
4) save the last export and import path used

### CODE REVIEW

**Comment by Marr11317** [comment]
This is the merge to mac version. I have not tested it under windows or linux yet.

**Comment by Jojo-Schmitz** [comment]
Actually I didn't need another PR (but it doesn't harm either), just wanmted yopu to be able to check whether it works on Mac (which I can't test)

File: batch_convert.qml [hunk 1]
```diff
@@ -31,12 +31,12 @@ MuseScore {
...
-  //id: mscoreSettings
-  //category: "ui/application"
-  //property var globalStyle
-  //}
+  
+  Settings {
+    id: mscorePathsSettings
+    category: "application/paths"
+    property var myScores
+    }
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: Not quite sure, but I seem to remember that this part had been commented out in the mac branch for a reason, like because it didn't work

File: batch_convert.qml [hunk 2]
```diff
@@ -46,19 +46,10 @@ MuseScore {
...
-  //else // MuseScoreStyleType.LIGHT_FUSION
-    //window.color = "lightgrey"
     }
 
-//Window {
     id: window
-  //visible: true
-  //title: qsTr("Choose Formats") // How?
-  //modality: Qt.ApplicationModal // behave like a dialog
-
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: These commented out things should rather be kept, I want the mac and the master branch be as similar as possible for easier handling og merge conflicte between the two

File: batch_convert.qml [hunk 3]
```diff
@@ -687,8 +744,8 @@ MuseScore {
           var excerpts = thisScore.excerpts
           for (var ex = 0; ex < excerpts.length; ex++) {
             if (excerpts[ex].partScore !== thisScore) // only list when not base score
-              excerptsList.push([excerpts[ex], fileBase, srcModifiedTime])
-            }
+              excerptsList.push([excerpts[ex], filePath, fileName, srcModifiedTime])
+            }
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: whitespace damage

File: batch_convert.qml [hunk 4]
```diff
@@ -651,33 +701,40 @@ MuseScore {
...
+             
+             if (res)
+               resultText.append("%1 → %2".arg(fileFullPath).arg(outFormats.extensions[j]))
+             else
+               resultText.append("Error: %1 → %2 not exported".arg(fileFullPath).arg(outFormats.extensions[j]))
+            } 
           else
-            resultText.append(qsTr("%1.%2 is up to date").arg(fileBase).arg(outFormats.extensions[j]))
+             resultText.append(qsTr("%1 is up to date").arg(fileFullPath))
           }
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: indent...

File: batch_convert.qml (hunk 1 above)
Comment (Marr11317) [reply batch_convert.qml]: Let me try it again.
I think the part that didn't work was setting the window color. [lines 50 to 53](https://github.com/Jojo-Schmitz/batch_export/pull/6/files/2516b695c61cf627d67d1ece68725ac42da2db8f#diff-8108c8b073573ef2b164b9f1f4873723L50)
But anyway the theme is followed by default.
I'll tell you monday afternoon.

File: batch_convert.qml [hunk 5]
```diff
@@ -31,12 +31,12 @@ MuseScore {
       Qt.quit()
       }
     }
-
-//Settings {
-  //id: mscoreSettings
-  //category: "ui/application"
-  //property var globalStyle
-  //}
+
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: whitespace damage

File: batch_convert.qml [hunk 6]
```diff
@@ -763,7 +837,8 @@ MuseScore {
       } else if (fileList.length > 0) {
         // if we found files, start timer do process them
         processTimer.running = true
-      } else {
+      }
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: indent

File: batch_convert.qml [hunk 7]
```diff
@@ -774,20 +849,24 @@ MuseScore {
...
 
     // set folder and filter in FolderListModel
-    files.folder = fileDialog.folder
+    files.folder = sourceFolderDialog.folder
 
     if (traverseSubdirs.checked) {
       files.showDirs = true
       files.showFiles = true
-    } else {
+    }
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: indent

File: batch_convert.qml (hunk 1 above)
Comment (Marr11317) [reply batch_convert.qml]: Ok, doesn't work on mac. Works on windows though... I guess this is why this is the mac branch :)

**Comment by Marr11317** [comment]
had to remove the setting to save the last opened time as it didn't work on mac.

**Comment by Marr11317** [comment]
> had to remove the setting to save the last opened time as it didn't work on mac.

Actually, it looks like it's done by default. No idea why, though.

**Comment by Marr11317** [comment]
I'll correct the master branch when I get some time, but that might be in a month, when the exams are over. 
Do you need anything else on that branch?

**Comment by Marr11317** [comment]
once this will be done, I have a new branch which allows you for batch transposing

**Comment by Marr11317** [comment]
Screen Shot 2019-11-25 at 8 51 05 PM

File: batch_convert.qml [hunk 8]
```diff
@@ -651,33 +699,40 @@ MuseScore {
...
           // modifiedTime() will return 0 for non-existing files
-          file.source = targetFile
-
           // if src is newer than existing write this file
-          if (srcModifiedTime > file.modifiedTime()) {
-            var res = writeScore(thisScore, targetFile, outFormats.extensions[j])
-            resultText.append("%1 → %2".arg(fileName).arg(outFormats.extensions[j]))
-            }
+          if (srcModifiedTime > fileScore.modifiedTime()) {
+             var res = writeScore(thisScore, fileScore.source, outFormats.extensions[j])
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: Still an indent issue here, but I'll fix it myself

**Comment by Jojo-Schmitz** [comment]
Looking forward to the batch transpose PR ;-) Try to make sure to 'recycle' strings from MuseScore for that, from its Transpose dialog and use a similar set of options too

File: batch_convert.qml [hunk 9]
```diff
@@ -340,14 +340,22 @@ MuseScore {
           } // RowLayout
         CheckBox {
           id: exportExcerpts
-          text: /*qsTr("Export linked parts")*/ qsTranslate("action", "Export parts")
+          text: /*qsTr("Export linked parts")*/ qsTranslate("action", "Export Parts")
```
Comment (Jojo-Schmitz) [inline batch_convert.qml]: I had to revert this change, this string gets 'recycled' from MuseScore, so need to follow the spelling there in order to get translated properly

**Comment by Marr11317** [comment]
Using the same set of options would take more time than exposing the transpose dialog to the plugin API, waiting for the PR to be merged, and using it only then, so I won't do it now, but until it gets done, some people might be happy to use the current version.
I just recommended it [here](https://musescore.org/en/node/20583#comment-963136), but if you don't like people using a fork of your plugin and not the original, just tell me :)

**Comment by Marr11317** [comment]
It works perfectly for me...
